client/src/modules/reports/generate/stock_exit/stock_exit.config.js

```diff
@@ -0,0 +1,83 @@
+angular.module('bhima.controllers')
+  .controller('stock_exitController', InventoryFileConfigController);
+
+InventoryFileConfigController.$inject = [
+  '$sce', 'NotifyService', 'BaseReportService', 'AppCache', 'reportData', '$state',
+  'LanguageService',
+];
+
+function InventoryFileConfigController($sce, Notify, SavedReports, AppCache, reportData, $state, Languages) {
+  const vm = this;
+  const cache = new AppCache('configure_stock_exit_report');
+  const reportUrl = 'reports/stock/exit';
+
+  // default values
+  vm.includePatientExit = 1;
+  vm.previewGenerated = false;
+
+  // check cached configuration
+  checkCachedConfiguration();
+
+  vm.onSelectDepot = depot => {
+    vm.depot = depot;
+  };
+
+  vm.clear = key => {
+    delete vm[key];
+  };
+
+  vm.clearPreview = () => {
+    vm.previewGenerated = false;
+    vm.previewResult = null;
+  };
+
+  vm.preview = form => {
+    if (form.$invalid) { return 0; }
+
+    const params = {
+      depotUuid : vm.depot.uuid,
+      dateFrom : vm.dateFrom,
+      dateTo : vm.dateTo,
+      includePatientExit : vm.includePatientExit,
+      includeServiceExit : vm.includeServiceExit,
+      includeDepotExit : vm.includeDepotExit,
+      includeLossExit : vm.includeLossExit,
+      showDetails : vm.showDetails,
+    };
+
+    // update cached configuration
+    cache.reportDetails = angular.copy(params);
+
+    const options = {
+      params,
```

Comment:
Why do you include the params as a JSON instead of just using HTTP's parameter serializing?  Is there a particular reason?

If not, I propose you do this:
```js
const options = angular.merge(params, { lang : Lanuages.key });
```
... which means that you can remove the `JSON.parse(req.query.params)` command from server because it will automatically be parsed by express's middleware.